code/game/objects/items/weapons/rpd.dm

```diff
@@ -0,0 +1,303 @@
+	list("key1" = "Bin",			"key2" = "6",	"key3" = "1",	"key4" = "condisposal"),
+	list("key1" = "Outlet",			"key2" = "7",	"key3" = "4",	"key4" = "outlet"),
+	list("key1" = "Chute",			"key2" = "8",	"key3" = "4",	"key4" = "intake"))
+
+//Procs
+
+/obj/item/weapon/rpd/proc/Activaterpd(soundfile, delay)
+	playsound(loc, soundfile, 50, 1)
+	if(prob(15))
+		spark_system.start()
+	if(delay)
+		cooldown = 1
+		spawn(spawndelay)
+		cooldown = 0
+	return
+
+/obj/item/weapon/rpd/proc/Checkifbent(x)
+	if(x in list(1,3,25,27))
+		return 1
+	else
+		return 0
+
+//This proc puts each necessary icon into the initial user's cache for later use. It then sets initialised = 1 to avoid calling itself unnecessarily.
+/obj/item/weapon/rpd/proc/Initialiseicons()
+	var/F
+	var/I
+	var/list/filenames
+	var/disposalsfile = "icons/obj/pipes/disposal.dmi"
+	var/pipefile = "icons/obj/pipe-item.dmi"
+	for(I in pipelist)
+		if(I["key6"] == "1")//Bendy pipe icons
+			filenames += list(
+			list(pipefile, I["key5"], I["key5"] + "-southeast.png", SOUTHEAST),
+			list(pipefile, I["key5"], I["key5"] + "-southwest.png", SOUTHWEST),
+			list(pipefile, I["key5"], I["key5"] + "-northwest.png", NORTHWEST),
+			list(pipefile, I["key5"], I["key5"] + "-northeast.png", NORTHEAST))
+			continue
+		if(I["key4"] == "2" || I["key4"] == "4") //Dual state icons
+			filenames += list(
+			list(pipefile, I["key5"], I["key5"] + "-north.png", NORTH),
+			list(pipefile, I["key5"], I["key5"] + "-east.png", EAST))
+		if(I["key4"] == "4")//All other pipes that need icons
+			filenames += list(
+			list(pipefile, I["key5"], I["key5"] + "-south.png", SOUTH),
+			list(pipefile, I["key5"], I["key5"] + "-west.png", WEST))
+	for(I in dpipelist)
+		if(I["key3"] == "2" || I["key3"] == "4")
+			filenames += list(
+			list(disposalsfile, I["key4"], I["key4"] + "-north.png", NORTH),
+			list(disposalsfile, I["key4"], I["key4"] + "-east.png", EAST))
+		if(I["key3"] == "4")
+			filenames += list(
+			list(disposalsfile, I["key4"], I["key4"] + "-south.png", SOUTH),
+			list(disposalsfile, I["key4"], I["key4"] + "-west.png", WEST))
+	for(F in filenames)
+		var/icon/iconfile = icon(F[1], icon_state = F[2], dir = F[4])
+		usr << browse_rsc(iconfile, F[3])
+	initialised = 1
+	return
+
+/obj/item/weapon/rpd/proc/Manipulatepipes(subject, targetturf, atmosverb, disposalsverb)
+	if(istype(subject, /obj/item/pipe))
+		call(subject, atmosverb)()
+	else if(istype(subject, /obj/structure/disposalconstruct/))
+		call(subject, disposalsverb)()
+	else
+		return
+
+//NanoUI stuff
+
+/obj/item/weapon/rpd/attack_self(mob/user as mob)
+	ui_interact(user)
+
+/obj/item/weapon/rpd/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1, var/datum/topic_state/state = inventory_state)
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "rpd.tmpl", "[name]", 600, 650, state = state)
+		ui.open()
+		if(!initialised)
```

Comment:
Then they press the reset button. I agree it's a bit dumb but the alternative was to call the icons proc every time you open the RPD menu (unless you can think of something to make it not do that)